### DISCUSSION TITLE
Performance vs Flexibility

### DISCUSSION THREAD

**Discussion by maceligueta** [KratosMultiphysics/Kratos#8754] [Ideas]
# **Performance vs Flexibility**
@KratosMultiphysics/all: As a consequence of the [3rd Kratos Community Meeting](https://github.com/KratosMultiphysics/Kratos/discussions/8580) we open this independent discussion to prepare a more specific online meeting around the topic 'Performance vs Flexibility'.

The date for this meeting will be **26th May 2021 at 10:00 CET**.
[****** add meeting to my Google Calendar ******](https://calendar.google.com/event?action=TEMPLATE&tmeid=N243bWZ0a2VrZjU3NzU2ZW1zNmJlZWNnNHIgNW9kY2liY2FsYmY5bWQ1dDZlcTc5OW41cXNAZw&tmsrc=5odcibcalbf9md5t6eq799n5qs%40group.calendar.google.com)

During the Community Meeting, @pooyan-dadvand clearly stated that "before optimizing the code, use a profiler to know where the bottlenecks are". So we thought that we could start this specific meeting with a **short introduction to Vtune**, where a demo will be carried out in order to show the main capabilities of this great profiler which, luckily, is now free-to-use.

The plan is to extend then the meeting to an open discussion about where to (and where not to) optimize our applications and the Kratos Core but, please, **feel free to propose ideas on the topic here**, which can be added to the agenda.

@antonialarese @rolandwuechner @maceligueta

**Comment by pooyan-dadvand** [comment]
@maceligueta will you be the one giving the vtune introduction?

**Reply by maceligueta** [reply]
I can do that, yes. Though I am not an expert at all! If anyone else volunteers, he/she is **more than welcome** to do this presentation.

**Comment by maceligueta** [comment]
Just a reminder that this meeting is tomorrow morning.

**Comment by antonialarese** [comment]
[ *** LINK TO THE VIDEO OF THE MEETING (Brief introduction to Intel VTune) *** ](https://mediaspace.unipd.it/media/KRATOS+Performance+vs+flexibility+meeting+%28VTune+profiler%29/1_hmnu0sry)

**Reply by salvalatorre** [reply]
Thanks!!
